### PR TITLE
feat(alloc): policies of memory allocator

### PR DESCRIPTION
**Pull request by Chillax-0v0**

### CODE REVIEW

File: s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java [hunk 1]
```diff
@@ -50,6 +47,15 @@ public class ByteBufAlloc {
     public static final int STREAM_SET_OBJECT_COMPACTION_WRITE = 10;
     public static ByteBufAllocMetric byteBufAllocMetric = null;
 
+    /**
+     * The policy used to allocate memory.
+     */
+    private static ByteBufAllocPolicy policy = ByteBufAllocPolicy.UNPOOLED_HEAP;
```
Comment (ShadowySpirits) [inline s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java:53]: How could the user set the policy?

File: s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java [hunk 2]
```diff
@@ -28,11 +28,8 @@
 
 public class ByteBufAlloc {
     public static final boolean MEMORY_USAGE_DETECT = Boolean.parseBoolean(System.getenv("AUTOMQ_MEMORY_USAGE_DETECT"));
```
Comment (ShadowySpirits) [inline s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java:30]: How about adding this configuration to the policy?

File: s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java (hunk 1 above)
Comment (ShadowySpirits) [reply s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java:53]: And it is better to add the policy attribute for metrics "buffer_used_memory_size".

Comment (Chillax-0v0) [reply s3stream/src/main/java/com/automq/stream/s3/ByteBufAlloc.java:53]: > How could the user set the policy?

Call `ByteBufAlloc#setPolicy(ByteBufAllocPolicy)`.

**Review by ShadowySpirits — APPROVED** [review]

**Comment by Chillax-0v0** [comment]
redo in https://github.com/AutoMQ/automq-for-kafka/pull/938

**Pull request merged**